cdap-security/src/main/java/co/cask/cdap/security/authorization/AbstractAuthorizationEnforcer.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.security.authorization;
+
+import co.cask.cdap.api.Predicate;
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.proto.id.EntityId;
+import co.cask.cdap.proto.security.Action;
+import co.cask.cdap.proto.security.Principal;
+import co.cask.cdap.security.spi.authorization.AuthorizationEnforcer;
+import co.cask.cdap.security.spi.authorization.UnauthorizedException;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Abstract Class that implements common methods for the {@link AuthorizationEnforcer} interface.
+ */
+public abstract class AbstractAuthorizationEnforcer implements AuthorizationEnforcer {
+
+  static final Predicate<EntityId> ALLOW_ALL = new Predicate<EntityId>() {
+    @Override
+    public boolean apply(EntityId entityId) {
+      return true;
+    }
+  };
+
+  private final boolean securityEnabled;
+  private final boolean authorizationEnabled;
+
+  public AbstractAuthorizationEnforcer(CConfiguration cConf) {
+    this.securityEnabled = cConf.getBoolean(Constants.Security.ENABLED);
+    this.authorizationEnabled = cConf.getBoolean(Constants.Security.Authorization.ENABLED);
+  }
+
+  @Override
+  public void enforce(EntityId entity, Principal principal, Set<Action> actions) throws Exception {
+    if (!isSecurityAuthorizationEnabled()) {
+      return;
+    }
+    Set<Action> disallowed = new HashSet<>(actions.size());
+    for (Action action : actions) {
+      try {
+        enforce(entity, principal, action);
+      } catch (Exception ex) {
+        disallowed.add(action);
+      }
+    }
+    if (disallowed.size() > 0) {
```

Comment:
We want to tell the user which privileges they are missing, makes it easier to correct it.